packages/kit/src/core/dev/index.js

```diff
@@ -84,47 +84,54 @@ class Watcher extends EventEmitter {
 
 		this.server = await get_server(this.https, user_config, (req, res) => handler(req, res));
 
-		/**
-		 * @type {vite.ViteDevServer}
-		 */
-		this.vite = await vite.createServer({
-			...user_config,
+		/** @type {[any, string[]]} */
+		const [merged_config, conflicts] = deep_merge(user_config, {
 			configFile: false,
 			root: this.cwd,
 			resolve: {
-				...user_config.resolve,
 				alias: {
-					...(user_config.resolve && user_config.resolve.alias),
 					$app: path.resolve(`${this.dir}/runtime/app`),
 					$lib: this.config.kit.files.lib
 				}
 			},
 			plugins: [
-				...(user_config.plugins || []),
 				svelte({
 					extensions: this.config.extensions,
 					emitCss: !this.config.kit.amp
 				})
 			],
 			publicDir: this.config.kit.files.assets,
 			server: {
-				...user_config.server,
 				middlewareMode: true,
 				hmr: {
-					...(user_config.server && user_config.server.hmr),
 					...(this.https ? { server: this.server, port: this.port } : {})
 				}
 			},
 			optimizeDeps: {
-				...user_config.optimizeDeps,
 				entries: []
 			},
 			ssr: {
-				...user_config.ssr,
 				noExternal: get_no_external(this.cwd, user_config.ssr && user_config.ssr.noExternal)
 			}
 		});
 
+		conflicts.forEach((conflict) => {
```

Comment:
could we put the `print_vite_config_conflicts` method somewhere shared and use it here?

Reply:
Cool, how about a generic "print_config_conflicts" that might as well use the logger() I suppose. See update.